packages/@aws-cdk/aws-iam/lib/policy-document.ts

```diff
@@ -90,12 +91,21 @@ export class AccountPrincipal extends ArnPrincipal {
  * An IAM principal that represents an AWS service (i.e. sqs.amazonaws.com).
  */
 export class ServicePrincipal extends PolicyPrincipal {
-  constructor(public readonly service: string) {
+  /**
+   * @param scope   a Construct scoping the service principal, typically the resource that will cause the service
+   *                principal to perform actions this policy statement authorizes, or it's parent Construct.
+   * @param service the short name of the service (e.g: s3, sns, sqs, secretsmanager, ...)
+   */
+  constructor(private readonly scope: cdk.IConstruct, private readonly service: string) {
```

Comment:
Can we get some validation in here so that people don't accidentally pass the full domain name? (Or better yet, strip `.amazonaws.com` from the end if it's there, to "do the right thing" automatically).

Reply:
I could but frankly I'd rather rely on people testing their code properly.